src/jdk.jartool/share/classes/jdk/security/jarsigner/JarSigner.java

```diff
@@ -417,29 +417,30 @@ private static boolean parseBoolean(String name, String value) {
         /**
          * Gets the default digest algorithm.
          *
-         * @implNote This implementation returns "SHA-256". The value may
+         * @implNote This implementation returns "SHA-384". The value may
          * change in the future.
          *
          * @return the default digest algorithm.
          */
         public static String getDefaultDigestAlgorithm() {
-            return "SHA-256";
+            return "SHA-384";
         }
 
         /**
          * Gets the default signature algorithm for a private key.
-         * For example, SHA256withRSA for a 2048-bit RSA key, and
+         * For example, SHA384withRSA for a 3072-bit RSA key, and
          * SHA384withECDSA for a 384-bit EC key.
          *
          * @implNote This implementation makes use of comparable strengths
-         * as defined in Tables 2 and 3 of NIST SP 800-57 Part 1-Rev.4.
-         * Specifically, if a DSA or RSA key with a key size greater than 7680
-         * bits, or an EC key with a key size greater than or equal to 512 bits,
+         * as defined in Tables 2 and 3 of NIST SP 800-57 Part 1-Rev.5 as
+         * well as NIST recommendations as appropriate.
+         * Specifically, if a DSA or RSA key with a key size no less than 7680
+         * bits, or an EC key with a key size no less than 512 bits,
          * SHA-512 will be used as the hash function for the signature.
```

Comment:
In this javadoc, SHA-512 for 7680-bit key (7680 is no less than 7680).

Reply:
Right, there are a few places which this is documented. Code and doc aren't closely coupled together plus changed course a few times... I will fix this and double check other files. Thanks!